File: backend/src/main/kotlin/de/predic8/openapicatalogbackend/model/entity/document/Report.kt
package de.predic8.openapicatalogbackend.model.entity.document

import de.predic8.openapicatalogbackend.model.dto.document.ReportDto
import de.predic8.openapicatalogbackend.model.entity.AEntity
import jakarta.persistence.CascadeType.ALL
import jakarta.persistence.Entity
import jakarta.persistence.Lob
import jakarta.persistence.OneToMany

@Entity
class Report(@Lob var diffReport: String? = null, @OneToMany(cascade = [ALL], mappedBy = "report", orphanRemoval = true) var lintReports: MutableList<LintReport> = mutableListOf()) : AEntity() {
    constructor(dto: ReportDto) : this(diffReport = dto.diffReport?.toString())
}

File: frontend/src/model/document/Report.ts
import AEntity from 'src/model/AEntity'
import type IReportDiff from 'src/model/document/IReportDiff'
import type LintReport from 'src/model/document/LintReport'

export default class Report extends AEntity {
  diffReport?: IReportDiff
  lintReports: LintReport[] = []
}
